opens asset links in a new tab on the backfills page

braunjj opened the pull request:
## Summary & Motivation
Makes asset links on the backfills overview tab open in a new tab so its easier to review each one and not lose your place in the app.

### Links in the Asset Name column now open in a new tab
<img width="1474" height="651" alt="image" src="https://github.com/user-attachments/assets/b68fef56-3342-4818-8de4-8bd4c28a82b4" />

salazarm reviewed (dismissed):
We should fix this to be a regular link that can be opened in a new tab or in the same tab

braunjj commented:
> We should fix this to be a regular link that can be opened in a new tab or in the same tab

@salazarm It looks like we were using a button link component so that we could do an async call to handle certain partitioned assets differently, but I don't think this is _really_ necessary. I removed it and converted this component back to a basic <Link> that links to the asset detail overview. This makes it possible to command click to open in a new tab.

This feels like an ok tradeoff to me, but want to double check. WDYT?
